### PR TITLE
Let the initial date picker mode be selectable

### PR DESCRIPTION
**Pull request by xster**
Let showDatePicker have an initial year/day mode argument.

Will add tests if people like it.

### CODE REVIEW

**Comment by xster** [comment]
#11610

File: packages/flutter/test/material/date_picker_test.dart [hunk 1]
```diff
@@ -283,6 +298,17 @@ void main() {
     });
   });
 
+  testWidgets('Can select initial date picker mode', (WidgetTester tester) async {
+    initialDatePickerMode = DatePickerMode.year;
+    await preparePicker(tester, (Future<DateTime> date) async {
+      await tester.pump();
+      // 2018 wouldn't be available if the year picker wasn't showing.
```
Comment (collinjackson) [inline packages/flutter/test/material/date_picker_test.dart]: Are you sure this is true? The month picker shows the year in two places (the button that switches to year mode, and the text showing the picked date)

Comment (xster) [inline packages/flutter/test/material/date_picker_test.dart]: Added another comment. The initial year is 2016. 2018 wouldn't on screen.

**Comment by collinjackson** [comment]
lgtm

File: packages/flutter/lib/src/material/date_picker.dart [hunk 2]
```diff
@@ -831,6 +845,10 @@ typedef bool SelectableDayPredicate(DateTime day);
 /// the days to enable for selection. If provided, only the days that
 /// [selectableDayPredicate] returned true for will be selectable.
 ///
+/// An optional [initialDatePickerMode] argument can be used to display the
+/// date picker initially in the year or month+day picker mode. It defaults
+/// to month+day, but cannot be null.
```
Comment (Hixie) [inline packages/flutter/lib/src/material/date_picker.dart]: why "but"?

File: packages/flutter/lib/src/material/date_picker.dart [hunk 3]
```diff
@@ -26,7 +26,18 @@ import 'material.dart';
...
+/// Date picker UI mode for either showing a list of available years or a
+/// monthly calendar.
+///
+/// Also see:
+///
+///  * <https://material.io/guidelines/components/pickers.html#pickers-date-pickers>
+enum DatePickerMode {
+  /// Show a date picker UI for choosing a month and day.
+  day,
+  /// Show a date picker UI for choosing a year.
```
Comment (Hixie) [inline packages/flutter/lib/src/material/date_picker.dart]: nit: blank line after "day,"

File: packages/flutter/lib/src/material/date_picker.dart [hunk 4]
```diff
@@ -26,7 +26,18 @@ import 'material.dart';
 import 'theme.dart';
 import 'typography.dart';
 
-enum _DatePickerMode { day, year }
+/// Date picker UI mode for either showing a list of available years or a
+/// monthly calendar.
```
Comment (Hixie) [inline packages/flutter/lib/src/material/date_picker.dart]: This should link to the API that uses this enum.

**Pull request merged**